api/options.c

```diff
@@ -270,7 +270,7 @@ nlopt_result
 NLOPT_STDCALL nlopt_set_lower_bounds(nlopt_opt opt, const double *lb)
 {
      nlopt_unset_errmsg(opt);
-     if (opt && (opt->n == 0 || lb)) {
+     if (opt && (opt->n > 0) && lb) {
```

Comment:
The point is that you can skip the `memcpy` whenever `opt->n == 0`, regardless of whether the pointers are non-NULL, so you might as well check `n > 0` before `memcpy` if you are going to insert a check here.

(Leave  the original`if (opt && (opt->n == 0 || lb)) {` as-is, since it was correct.)